File: src/GameServerManager/Common/Services/GameModeService.cs
using Crpg.Domain.Entities.Servers;

namespace Crpg.GameServerManager.Common.Services;

internal static class GameModeService
{
    private static readonly Dictionary<GameModeAlias, GameMode> GameModeByInstanceAlias = new()
    {
        { GameModeAlias.A, GameMode.CRPGBattle },
        { GameModeAlias.B, GameMode.CRPGConquest },
        { GameModeAlias.C, GameMode.CRPGDuel },
        { GameModeAlias.E, GameMode.CRPGDTV },
        { GameModeAlias.D, GameMode.CRPGSkirmish },
        { GameModeAlias.F, GameMode.CRPGTeamDeathmatch },
        { GameModeAlias.Z, GameMode.CRPGUnknownGameMode },
        { GameModeAlias.G, GameMode.CRPGCaptain },
    };

    private static readonly Dictionary<GameMode, GameModeAlias> AliasByGameMode =
       GameModeByInstanceAlias.ToDictionary(kvp => kvp.Value, kvp => kvp.Key);

    public static GameMode GetGameModeByInstanceAlias(GameModeAlias alias)
    {
        return GameModeByInstanceAlias[alias];
    }

    public static GameModeAlias GetInstanceAliasByGameMode(GameMode mode)
    {
        return AliasByGameMode.TryGetValue(mode, out var alias) ? alias : GameModeAlias.Z;
    }
}

File: src/WebUI/app/services/game-mode-service.ts
import type { GameMode } from '~/models/game-mode'

import { GAME_MODE } from '~/models/game-mode'

export const gameModeToIcon: Record<GameMode, string> = {
  [GAME_MODE.CRPGBattle]: 'game-mode-battle',
  [GAME_MODE.CRPGConquest]: 'game-mode-conquest',
  [GAME_MODE.CRPGDTV]: 'game-mode-dtv',
  [GAME_MODE.CRPGDuel]: 'game-mode-duel',
  [GAME_MODE.CRPGTeamDeathmatch]: 'game-mode-teamdeathmatch',
  [GAME_MODE.CRPGCaptain]: 'game-mode-captain',
  [GAME_MODE.CRPGUnknownGameMode]: '',
  [GAME_MODE.CRPGSiege]: '',
  [GAME_MODE.CRPGSkirmish]: '',
}

export const rankedGameModes: GameMode[] = [GAME_MODE.CRPGBattle, GAME_MODE.CRPGDuel]

export const checkIsRankedGameMode = (gameMode: GameMode) => rankedGameModes.includes(gameMode)
